test/cmd/core.sh

```diff
@@ -992,6 +992,32 @@ __EOF__
   # Post-condition: Only the default kubernetes services exist
   kube::test::get_object_assert services "{{range.items}}{{$id_field}}:{{end}}" 'kubernetes:'
 
+  ### Create deployent and service
+  # Pre-condition: no deployment exists
+  kube::test::wait_object_assert deployment "{{range.items}}{{$id_field}}:{{end}}" ''
+  # Command
+  kubectl run testmetadata --image=nginx --replicas=2 --port=80 --expose --service-overrides='{ "metadata": { "annotations": { "zone-context": "home" } } } '
+  # Check result
+  kube::test::get_object_assert deployment "{{range.items}}{{$id_field}}:{{end}}" 'testmetadata:'
+  kube::test::get_object_assert services "{{range.items}}{{$id_field}}:{{end}}" 'kubernetes:testmetadata:'
+
+  ### Expose deployment as a new service
+  # Command
+  kubectl expose deployment testmetadata  --port=1000 --target-port=80 --type=NodePort --name=exposemetadata --overrides='{ "apiVersion": "v1" , "kind": "Service" , "metadata": { "annotations": { "zone-context": "work" } } } '
+  # Check result
+  kube::test::get_object_assert services "{{range.items}}{{$id_field}}:{{end}}" 'exposemetadata:kubernetes:testmetadata:'
```

Comment:
verify the resulting object has the overridden annotation

Reply:
@liggitt
     Add case to check the annotation's content instead of the service's list.